src/debugger/awsSamDebugger.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+const SAM_APP_DEBUG_TYPE = 'aws-sam'
+const SAM_APP_REQUEST_TYPES = new Set<string>(['direct-invoke'])
+const SAM_APP_TARGET_TYPES = new Set<string>(['template', 'code'])
+
+export interface AwsSamDebugConfiguration extends vscode.DebugConfiguration {
+    readonly invokeTarget: {
```

Comment:
1. I'd prefer to see all of the inner structures defined as separate interfaces. This will make it easier to pass those substructures around in code, and to do things like what is proposed in...
2. when target is `code`, invokeTarget has the fields `projectRoot` and `lambdaHandler`. This could be handled as one interface with all optional fields, or as a base interface (containing target) and two separate derived interfaces.